src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -669,5 +676,16 @@ private bool MeetsConstraints(NuGetVersion newVersion, InstalledPackageReference
 
             return result;
         }
+
+        private SourceRepository GetOrAddSourceRepository(IEnumerable<Lazy<INuGetResourceProvider>> providers, PackageSource packageSource)
```

Comment:
If you initialize the sources before the code starts running in parallel, then you don't need to use a concurrent data structure. For example, immediately after the code determines we're not in an offline-only command: https://github.com/NuGet/NuGet.Client/blob/3094c4e5f28b345de5a00e6a43f596d4b5eb72cf/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs#L112-L116

Reply:
@zivkan 
Thank you for your review. I implemented your suggestion, please check again.